Sources/SwiftInspectorCommands/TypeLocationCommand.swift

```diff
@@ -0,0 +1,80 @@
+// Created by Michael Bachand on 3/28/20.
+//
+// Copyright (c) 2020 Francisco Diaz
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import ArgumentParser
+import Foundation
+import SwiftInspectorCore
+
+final class TypeLocationCommand: ParsableCommand {
+  static var configuration = CommandConfiguration(
+    commandName: "type-location",
+    abstract: "Finds the location of a type"
+  )
+
+  @Option(help: nameArgumentHelp)
+  var name: String
+
+  @Option(help: "The absolute path to the file to inspect")
+  var path: String
+
+  /// Runs the command
+  func run() throws {
+    let cachedSyntaxTree = CachedSyntaxTree()
+
+    let analyzer = TypeLocationAnalyzer(typeName: name, cachedSyntaxTree: cachedSyntaxTree)
+    let fileURL = URL(fileURLWithPath: path)
+    let typeLocations = try analyzer.analyze(fileURL: fileURL)
+
+    if !typeLocations.isEmpty {
+      print(typeLocations.map { $0.outputString() }.joined(separator: "\n"))
+    }
+  }
+
+  /// Validates if the arguments of this command are valid
+  func validate() throws {
+    guard !name.isEmpty else {
+      throw InspectorError.emptyArgument(argumentName: "--name")
+    }
+    guard !path.isEmpty else {
+      throw InspectorError.emptyArgument(argumentName: "--path")
+    }
+    var isDir: ObjCBool = false
+    guard FileManager.default.fileExists(atPath: path, isDirectory: &isDir) && !isDir.boolValue else {
+      throw InspectorError.invalidArgument(argumentName: "--path", value: path)
+    }
+  }
+}
+
+extension LocatedType {
+
+  func outputString() -> String {
+    return "\(indexOfStartingLine) \(indexOfEndingLine)"
+  }
+}
+
+// We recognize that a protocol may not strictly be a type, but we're OK with cutting that corner
+// for now...
```

Comment:
I'm not sure I follow this comment. Can you clarify?

Reply:
Consider this output. Swift doesn't seem to call a protocol a type.

<img width="437" alt="Screen Shot 2020-04-04 at 2 18 47 PM" src="https://user-images.githubusercontent.com/1791049/78461579-404e0300-767f-11ea-8e60-d2b8f5a60107.png">

Reply:
@bachand are you referring to `ArgumentHelp`? `ArgumentHelp` it's a struct, not a type.

Reply:
Ah sorry let me be more clear.

The name of this command is "type location". So it finds the location of _types_.

I am not sure that Swift considers protocols to be "types". In Swift, a structure or class or enum is clearly a type. But I'm not sure the language officially thinks that a protocol is a type.

What I'm trying to communicate with the code output is that invoking `.self` on a protocol functions differently than invoking `.self` on a class, enum, or structure. This output may be more clear.

<img width="442" alt="Screen Shot 2020-04-04 at 3 52 07 PM" src="https://user-images.githubusercontent.com/1791049/78462965-4dbdba00-768c-11ea-846e-d479407d6c76.png">

So anyway the reason that I added this comment was to explain that even though the command is called "type location" it is able to find protocols too, even if Swift doesn't consider a protocol to actually be a type.

Does that make sense?

Happy to remove or make more clear, if you have suggestions.

Reply:
@bachand Oh gotcha! I think you're conflating Types with Meta Types. The things you're referring to are meta types, [see this](https://docs.swift.org/swift-book/ReferenceManual/Types.html#ID455).

`Foo` in your example is the type of your protocol, while `Foo.Protocol` is the meta type.

Reply:
OK I removed this comment https://github.com/fdiaz/SwiftInspector/pull/13/commits/be073028222b98aba43c0b89d4c8aa243e08f655